flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -354,24 +350,30 @@ void ArrayCopyAnalysis::arrayMentions(
 
   // Process the worklist until done.
   while (!queue.empty()) {
-    auto *operand = queue.pop_back_val();
-    auto *owner = operand->getOwner();
-    if (!owner)
-      continue;
+    mlir::OpOperand *operand = queue.pop_back_val();
+    mlir::Operation *owner = operand->getOwner();
     auto structuredLoop = [&](auto ro) {
       if (auto blockArg = ro.iterArgToBlockArg(operand->get())) {
-        auto arg = blockArg.getArgNumber();
-        auto output = ro.getResult(ro.finalValue() ? arg : arg - 1);
+        int64_t arg = blockArg.getArgNumber();
```

Comment:
Is an argument number really (and always) 64-bit? That's a little weird. If we have more than 2^32 arguments, I suspect we're in a lot of trouble...